p2p/dial.go

```diff
@@ -388,6 +389,9 @@ func (d *dialScheduler) checkDial(n *enode.Node) error {
 	if n.ID() == d.self {
 		return errSelf
 	}
+	if n.IP() != nil && n.TCP() == 0 {
```

Comment:
Why do you need to check IP nil-ness?

Reply:
I already explained this in the earlier GH comment, but will add a code comment.

Reply:
Oh, sorry, RTFM